suprb/optimizer/rule/es/es.py

```diff
@@ -105,6 +128,26 @@ def _optimize(self, X: np.ndarray, y: np.ndarray, initial_rule: Rule, random_sta
                 warnings.warn("No valid children were generated during this iteration.", UserWarning)
                 continue
 
+            """Use the adapted version of the 1/5th-Rule as introduced in:
+               https://direct.mit.edu/evco/article-abstract/9/2/147/898/Convergence-in-Evolutionary-Programs-with-Self 
+               proportion: float 
+                    How many of the lmbda children create are fitter than their parent
+            """
+            if self.adaptive_sigma:
+                matching_type = elitist.match.__class__.__name__
+                fitter_children = sum([1 for child in children if child.fitness_ > elitist.fitness_])
+                proportion = fitter_children / self.lmbda
+
+                if proportion > 0.2:
+                    self.alter_sigma(self.adaption_rate, "/", matching_type, random_state)
+                elif 0.05 <= proportion < 0.2:
+                    self.alter_sigma(self.adaption_rate, "*", matching_type, random_state)
+                # Double the current mutation rate if proportion < 0.05 to escape any local minima
+                elif proportion < 0.05:
+                    self.alter_sigma(2, "*", matching_type, random_state)
+
+                self.mutation.sigma = np.clip(self.mutation.sigma, 0.001, 3)
```

Comment:
Is there a reason for this setting the range `[0.001, 3]` ?

Reply:
The clipping to 0.001 is done to prevent sigma from ever reaching 0 (would make further altering impossible)
Cliping to 3 was done to prevent sigma from exploding (reaching values larger > 1000). The value 3 was chosen as it also was the upper bound provided during tuning